src/cmd/controller/main.go

```diff
@@ -55,63 +89,190 @@ func init() {
+	if err != nil {
+		logger.With("error", err).Fatal("failed to initialize AWS session")
+	}
+	sqsClient := sqs.New(awsSession)
+	if sqsClient == nil {
+		logger.Fatal("failed to create SQS client")
+	}
+	asgClient := autoscaling.New(awsSession)
+	if asgClient == nil {
+		logger.Fatal("failed to create ASG client")
+	}
+	ec2Client := ec2.New(awsSession)
+	if ec2Client == nil {
+		logger.Fatal("failed to create EC2 client")
+	}
+
+	nodeGetter, err := node.NewGetter(kubeClient)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create node getter")
+	}
+	nodeNameGetter, err := nodename.NewGetter(ec2Client)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create node name getter")
+	}
+
+	asgTerminateEventV1Parser, err := asgterminateeventv1.NewParser(asgClient)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create ASG instance-terminate lifecycle event v1 parser")
+	}
+	asgTerminateEventV2Parser, err := asgterminateeventv2.NewParser(asgClient)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create ASG instance-terminate lifecycle event v2 parser")
+	}
+	sqsMessageParser, err := terminator.NewSqsMessageParser(event.NewParser(
+		asgTerminateEventV1Parser,
+		asgTerminateEventV2Parser,
+		rebalancerecommendationeventv0.NewParser(),
+		scheduledchangeeventv1.NewParser(),
+		spotinterruptioneventv1.NewParser(),
+		statechangeeventv1.NewParser(),
+	))
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create SQS message parser")
+	}
+
+	terminatorGetter, err := terminator.NewGetter(kubeClient)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create terminator getter")
 	}
 
-	if err = (&controllers.TerminatorReconciler{
-		Client: mgr.GetClient(),
-		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
-		ctrlLogger.With("error", err).Fatalw("unable to create controller")
+	sqsMessageClient, err := sqsmessage.NewClient(sqsClient)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create SQS message client")
+	}
+	terminatorSqsClientBuilder, err := terminator.NewSqsClientBuilder(sqsMessageClient)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create terminator SQS message client builder")
+	}
+
+	cordonDrainerBuilder, err := kubectlcordondrainer.NewBuilder(
+		clientSet,
+		kubectlcordondrainer.DefaultCordoner,
+		kubectlcordondrainer.DefaultDrainer,
+	)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create kubectl cordon/drain client")
+	}
+	terminatorCordonDrainerBuilder, err := terminator.NewCordonDrainerBuilder(cordonDrainerBuilder)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create terminator cordon/drain client")
+	}
+
+	rec := terminator.Reconciler{
+		Name:                 "terminator",
+		RequeueInterval:      time.Duration(10) * time.Second,
+		NodeGetter:           nodeGetter,
+		NodeNameGetter:       nodeNameGetter,
+		SqsClientBuilder:     terminatorSqsClientBuilder,
```

Comment:
in general initialisms and acronyms should have consistent casing: https://github.com/golang/go/wiki/CodeReviewComments#initialisms 

For some reason go made a distinction between human written code (which should use consistent casing) and machine generated code (which doesn't have to use consistent casing).